### PR TITLE
[TUI] Add minimal TUI with functioning tabs & CSS (1)

### LINKED ISSUE #185: Create empty main window with tabs (1)
The first step in the TUI is to create a main window app page that will hold a tab widget, with pages 'Create' and 'Transfer'. The aim is for switching tabs to switch the page in the area underneath (image below). It seems Textual has already combined Tabs and ContentSwitcher into a [TabbedContent](https://textual.textualize.io/widgets/tabbed_content/) widget which seems to be exactly what we want.

We can include simple text on each content window that just says 'create' and 'transfer' so we know the tab switching is working well.  Once this is done we can start building the 'Create' page.

Even at this early stage it is probably worth separating the css out into separate files (even if these are currently blank). We can follow the example at [ContentSwitcher](https://textual.textualize.io/widgets/content_switcher/#__tabbed_1_2) (where `CSS_PATH = "content_switcher.tcss"` and they show the example .css file). We can keep this blank or add one field to check it is working. If your happy with it, we can go with a structure such as:

```
.
└── datashuttle/
    ├── pyproject.toml
    ├── ....
    └── datashuttle/
        ├── configs
        ├── ...
        └── tui/
            ├── css/
            │   └── content_switcher.css
            └── view/
                └── app.py  # contains the app / content switcher
```
We can figure out exactly what to call these folders / files later when their function is clearer.

Image ( for now, lets ignore the 'Setup' tab, and the top bar with datashuttle and the minimize / restore / close buttons).
mainwindow 

To summarize, we can:

1) Create an initial app with TabbedContent widget with tabs 'Create' and 'Transfer'
2) add the text 'create' and 'transfer' to the content area, so that switching tabs switches the text.
3) create a `.css` file for the TabbedContent widget and add one field (e.g. the first feild from the example) to check it works.

Please let me know if there are any questions or anything is not clear!

### PR DESCRIPTION
**Pull request by b-peri**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/neuroinformatics-unit/.github/blob/main/CONTRIBUTING.md).

## Description

**What is this PR**

- [ ] Bug fix
- [X] Addition of a new feature
- [ ] Other

**What does this PR do?**
This PR adds a minimal TUI app built using the Textual package. While the window currently contains only 2 (largely empty) tabs, our intention is that this will ultimately form the skeleton of the DataShuttle graphical user interface.

## References
Closes #185

## How has this PR been tested?
At the current moment, the minimal TUI runs independently from the DataShuttle core software. Testing has been done by simply running the app via the terminal.

## Is this a breaking change?
No.

## Does this PR require an update to the documentation?
I don't think the app has reached a state where a change to documentation is warranted quite yet.

## Checklist:

- [X] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [X] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by adamltyson** [comment]
Hey @b-peri I wanted to play with this, so I added a couple of things to make it easier to install and run on linux/macOS. I'll let @JoeZiminski review this, but it's looking good!

**Comment by b-peri** [comment]
> Hey @b-peri I wanted to play with this, so I added a couple of things to make it easier to install and run on linux/macOS. I'll let @JoeZiminski review this, but it's looking good!

No problem @adamltyson, thanks! I've just put this in the issue (#185) as well but if you want to play around with the CSS bear in mind that Textual supports live CSS editing (See https://textual.textualize.io/guide/devtools/)

**Review by JoeZiminski — CHANGES_REQUESTED** [review]
Thanks @b-peri, this is fantastic! Just some minor comments
It would also be nice to add light mode at this stage for testing (I have a hard time with dark mode, but am a minority :sweat_smile: ). For now we could implement this as a CTRL+D shortcut as described in [this tutorial](https://textual.textualize.io/tutorial/). It seems fairly straightforward, the main logic is a switch `self.dark = not self.dark)`. 

- [ ] Implement dark mode switchable with CTRL+D

Down the line we will want to implement a settings page where things like view options can be changed (this could include a light / dark mode switch). Ideally this would be switchable from a menubar’s drop-down menu, however [this is not yet implemented in textual](https://github.com/Textualize/textual/issues/1092). We could try and implement something ourselves, similar to [this](https://github.com/Textualize/textual/discussions/1713) mixed with a [screen](https://textual.textualize.io/guide/screens/) that could pop-up a menu window that covers the rest of the screen. Alternatively it might be easier just to add a ‘Settings’ tab, one section of which is ‘View’.
Nonetheless (unless you are particularly interested in doing this now) we can think about this later when we have more of the core functionality build up. I can make an issue for this.

Cheers! Let me know if you have any questions, if you do not agree with a suggestion feel free to discuss why its not a good idea!

File: datashuttle/tui/view/app.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class MyApp(App):
+    TITLE = "DataShuttle"
+
+    tui_path = Path(__file__).parents[1]
+    CSS_PATH = tui_path / "css" / "tab_content.tcss"
+
+    def compose(self) -> ComposeResult:
+        yield Header()
+        # yield Label("Do you love DataShuttle?", id = "question")
```
[the remark this answers is not in the document]
Comment (b-peri) [reply datashuttle/tui/view/app.py:17]: Haha sorry about this one! I pasted this line from Textual's [App Basics guide page](https://textual.textualize.io/guide/app/#widgets) when I was playing around with CSS - it was just there for testing!

**Comment by JoeZiminski** [comment]
Thanks @b-peri! Looks great, we can merge this as-is into the new `tui_dev_branch`. I would suggest moving `# Compose window` into a docstring for the `compose()` function (+ a little more info), and removing the `#Set path to TCSS` and `'# Set key-bindings` comments as these are arguably self-evident from the code itself. However this is very minor / subjective and can be adressed on another PR. Looking good to go!

**Comment by JoeZiminski** [comment]
I forgot to mention for posterity, as discussed in #184 we will test at a more sensible time once a few more features are in place and we get user feedback.

**Comment by b-peri** [comment]
> Thanks @b-peri! Looks great, we can merge this as-is into the new `tui_dev_branch`. I would suggest moving `# Compose window` into a docstring for the `compose()` function (+ a little more info), and removing the `#Set path to TCSS` and `'# Set key-bindings` comments as these are arguably self-evident from the code itself. However this is very minor / subjective and can be adressed on another PR. Looking good to go!

Sounds good @JoeZiminski, thanks for the feedback! I've implemented your last few comments as part of #215, and so will close this PR now.

**Pull request merged**